contrib/inventory/foreman.py

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env python
+# vim: set fileencoding=utf-8 :
+#
+# Copyright (C) 2016 Guido Günther <agx@sigxcpu.org>
+#
+# This script is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with it.  If not, see <http://www.gnu.org/licenses/>.
+#
+# This is somewhat based on cobbler inventory
+
+from __future__ import print_function
+
+import argparse
+import copy
+import os
+import re
+import requests
+from requests.auth import HTTPBasicAuth
+import sys
+from time import time
+
+try:
+    import ConfigParser
+except ImportError:
+    import configparser as ConfigParser
+
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+
+class ForemanInventory(object):
+    config_paths = [
```

Comment:
This doesn't really matter since only a single instance of ForemanInventory is created in this script but config_paths should really be an instance attribute (created and given an initial value in ```__init__```) rather than a class attribute.  Class attributes exist a single time per class.  All instances of the class would share the same instance of that attribute.  Instance attributes exist, one per instance and thus can change independently.  Modifying config_paths later in the code is a tip-off that this should be an instance-attribute rather than a class-attribute.

Also, you can probably move the setting from an environment variable to be with this code when you do that.  It would seem to make sense to keep that all together.